core/src/components/toast/toast.tsx

```diff
@@ -275,7 +293,21 @@ export class Toast implements ComponentInterface, OverlayInterface {
 
     await this.delegateController.attachViewToDom();
 
-    await present<ToastPresentOptions>(this, 'toastEnter', iosEnterAnimation, mdEnterAnimation, this.position);
+    const { el, position } = this;
+    const anchor = this.getAnchorElement();
+    const animationPosition = getAnimationPosition(position, anchor, getMode(), el);
```

Comment:
The `getMode` call here will get the application mode, but it should get the component mode instead.

For example, if the app is in MD mode but the toast is in iOS mode then the toast currently does not appear. You can test this by setting the footerToast" mode to "ios".

Instead, we should call `getIonMode(this)` which will take into account the component's mode.